### ISSUE TITLE
Tuple struct does not respect skip_serializing

### ISSUE THREAD

**Issue by dtolnay** [serde-rs/serde#1048]
```rust
#[macro_use]
extern crate serde_derive;

extern crate serde;
extern crate serde_json;

#[derive(Serialize, Debug)]
struct S(usize, #[serde(skip)] usize);

fn main() {
    let s = S(0, 0);
    let j = serde_json::to_string(&s).unwrap();
    println!("{}", j);
}
```

```
[0,0]
```

Open question: whether a tuple struct with all but one skipped field should be serialized as a tuple struct of length 1 or a newtype struct.

**Comment by konstin** [comment]
Newtype, please. The current behavior causes  problems when using `PhantomData<T>`.

In my case I want to add another type to my custom url type, which can also be deserialized.

```rust
#[derive(Deserialize, Serialize)]
pub struct CustomUrl<T: DeserializeOwned>(
    #[serde(with = "url_serde")]
    pub Url,
    #[serde(skip)]
    PhantomData<T>
);
```

Is there any way to tell serde to treat the above as a newtype struct? The above code is currently failing as serde expects a tuple where there's actually a string.

**Comment by durka** [comment]
I hit this today trying to make a serializable enum with tuple struct variants containing unserializable fields. I can't do it because `#[skip]` is ignored.

**Comment by dtolnay** [comment]
Fixed in b4e51fcc7735ca3c60a7e70bb8d5ea2bf30edeea. It will continue to serialize as a tuple even if there is only one field. #1054 is tracking an attribute that would make types with a single serialized field serialize just like the underlying field.
